java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java

```diff
@@ -119,20 +121,10 @@ public void visitArrayAccessExpression(ArrayAccessExpressionTree tree) {
   }
 
   @Override
-  public void visitAssignmentExpression(AssignmentExpressionTree tree) {
-    if (stack.peek() != null) {
-      raiseIssue(tree);
-    }
-    stack.push(Tree.Kind.ASSIGNMENT);
-    super.visitAssignmentExpression(tree);
-    stack.pop();
-  }
-
-  @Override
   public void visitBinaryExpression(BinaryExpressionTree tree) {
-    Tree.Kind kind = getKind(tree);
     Tree.Kind peek = stack.peek();
-    if (peek != null && peek != kind && !isException(peek, kind)) {
+    Tree.Kind kind = getKind(tree);
+    if (TABLE.get(peek, kind) == Boolean.TRUE) {
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Change this comparison to use the equals method.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)